src/line-chart.js

```diff
@@ -201,10 +236,49 @@ dc.lineChart = function (parent, chartGroup) {
         return _chart.getColor.call(d, d.values, i);
     }
 
-    // Behavior of interpolator has changed in D3v4
-    var _interpolateWithTension = function () {
-        return typeof _interpolate.tension === 'function' ?
-            _interpolate.tension(_tension) : _interpolate;
+    // To keep it backward compatible, it covers multiple cases
+    // See https://github.com/dc-js/dc.js/issues/1376
+    // It will be removed when interpolate and tension are deprecated.
+    var _getCurveFn = function () {
+        var curve = null;
+
+        // _curve takes precedence
+        if (_curve) {
+            return _curve;
+        }
+
+        // Approximate the D3v3 behavior
+        if (typeof _interpolate === 'function') {
+            curve = _interpolate;
+        } else {
+            // If _interpolate is string
+            var mapping = {
+                'linear': d3.curveLinear,
+                'linear-closed': d3.curveLinearClosed,
+                'step': d3.curveStep,
+                'step-before': d3.curveStepBefore,
+                'step-after': d3.curveStepAfter,
+                'basis': d3.curveBasis,
+                'basis-open': d3.curveBasisOpen,
+                'basis-closed': d3.curveBasisClosed,
+                'bundle': d3.curveBundle,
+                'cardinal': d3.curveCardinal,
+                'cardinal-open': d3.curveCardinalOpen,
+                'cardinal-closed': d3.curveCardinalClosed,
+                'monotone': d3.curveMonotoneX
+            };
+            curve = mapping[_interpolate];
+        }
+
+        // Default value
+        if (!curve) {
+            curve = d3.curveLinear;
+        }
+
+        if (_tension && typeof curve.tension === 'function') {
+            curve = curve.tension(_tension);
```

Comment:
Please add a warning here if `_tension` was specified but the curve doesn't support it.

Reply:
While fixing also realized that 0 is a valid value for _tension, which would have gotten discarded.